test/jdk/java/lang/reflect/code/TestRemoveFinalVars.java

```diff
@@ -0,0 +1,86 @@
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.code.Block;
+import java.lang.reflect.code.CopyContext;
+import java.lang.reflect.code.Op;
+import java.lang.reflect.code.Value;
+import java.lang.runtime.CodeReflection;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.lang.reflect.code.op.CoreOps.FuncOp;
+import static java.lang.reflect.code.op.CoreOps.VarAccessOp.VarLoadOp;
+import static java.lang.reflect.code.op.CoreOps.VarAccessOp.VarStoreOp;
+import static java.lang.reflect.code.op.CoreOps.VarOp;
+
+/*
+ * @test
+ * @run testng TestRemoveFinalVars
+ */
+
+public class TestRemoveFinalVars {
+
+    @CodeReflection
+    static int f(boolean b) {
+        final int x = 8; // final var
+        int y = x + 2; // final var
+        int z = y + 3; // non final var
+        z++;
+        return y;
+    }
+
+    @Test
+    void test() {
+        FuncOp f = getFuncOp(this.getClass(),"f");
+        f.writeTo(System.out);
+
+        FuncOp f2 = f.transform(TestRemoveFinalVars::rmFinalVars);
+        f2.writeTo(System.out);
```

Comment:
We should assert on the resulting model or run through the interpreter. I suspect the result of your transformation would result in the same structure as the one produced by `SSA.transform`.

Reply:
Because we only remove final and effectively final variables, and SSA remove all variables, the result will be different.